onnxruntime/core/providers/cpu/controlflow/scan.h

```diff
@@ -19,7 +19,7 @@ namespace detail {
 struct DeviceHelpers {
   using ZeroData = std::function<common::Status(void* data, size_t size_in_bytes)>;
   using Transpose = std::function<common::Status(const std::vector<size_t>& permutations,
-                                                 const Tensor& input, Tensor& output)>;
+                                                 const Tensor& input, Tensor& output, const TensorShape* input_shape_override)>;
```

Comment:
>const TensorShape* input_shape_override [](start = 86, length = 39)

This doesn't need to be exposed here. The CPU implementation can set DeviceHelpers::Transpose to a lambda that plug in a nullptr when calling.

Reply:
Pushed a fix for this